src/type_checker.cpp

```diff
@@ -475,6 +504,21 @@ void TypeChecker::Visit(PostfixArithExprNode& postfix_expr) {
   postfix_expr.type = postfix_expr.operand->type->Clone();
 }
 
+void TypeChecker::Visit(RecordMemExprNode& mem_expr) {
+  mem_expr.expr->Accept(*this);
+  if (auto* record_type =
+          dynamic_cast<RecordType*>((mem_expr.expr->type).get())) {
+    if (record_type->IsMember(mem_expr.id)) {
+      mem_expr.type = record_type->MemberType(mem_expr.id)->Clone();
```

Comment:
Since the function `MemberType` already returns a rvalue unique pointer, we do not have to clone it again.
```suggestion
      mem_expr.type = record_type->MemberType(mem_expr.id);
```